benchmarking/benchmark.py

```diff
@@ -54,49 +54,101 @@ def get_times_skfp(X: np.ndarray, transformer_cls: type, **kwargs) -> np.ndarray
     return np.array(result).reshape((len(N_CORES), N_SPLITS))
 
 
-def save_results(
+LEGAL_PLOT_TYPES = ["time", "speedup", "fprints_per_second"]
+
+
+def save_plots(
     n_molecules: int,
     times: np.ndarray,
     title: str = "",
     save: bool = True,
+    plot_type: str = "time",
 ) -> None:
+    dir_name = plot_type
 
     X = n_molecules * np.linspace(0, 1, N_SPLITS + 1)[1:]
 
     plt.rcParams["font.size"] = 20
     fig = plt.figure(figsize=(15, 10))
     ax1 = fig.add_subplot()
     ax1.set_title(title)
-
-    for i, y in zip(N_CORES, times):
-        ax1.plot(X, y, label=f"our time - # cores: {i}")
-
-    ax1.set_ylabel("Time of computation [s]")
-    ax1.set_xlabel("Number of fingerprints")
+    ax1.set_xlabel("Number of molecules")
+
+    if plot_type == "time":
+        ax1.set_ylabel("Time of computation")
+        for i, y in zip(N_CORES, times):
+            ax1.plot(X, y, marker="o", label=f"# cores: {i}")
+    elif plot_type == "speedup":
+        ax1.set_ylabel("Speedup")
+        for i, y in zip(N_CORES[1:], times[1:]):
+            ax1.plot(X, times[0] / y, marker="o", label=f"# cores: {i}")
+    elif plot_type == "fprints_per_second":
+        ax1.set_ylabel("Fingerprints / s")
+        for i, y in zip(N_CORES, times):
+            ax1.plot(X, X / y, marker="o", label=f"# cores: {i}")
 
     ax1.set_xlim(n_molecules * 0.1, n_molecules * 1.1)
     ax1.set_ylim(bottom=0)
 
     plt.legend(loc="upper left", fontsize="14")
 
-    title = title.replace(" ", "_")
-    np.save(os.path.join(SCORE_DIR, f"{title}.npy"), times)
     fig.tight_layout()
 
     if save:
-        plt.savefig(os.path.join(PLOT_DIR, f"{title}.png"))
+        os.makedirs(os.path.join(PLOT_DIR, dir_name), exist_ok=True)
+        plt.savefig(os.path.join(PLOT_DIR, dir_name, f"{title}.png"))
     else:
         plt.show()
 
     plt.close(fig)
 
 
+def save_combined_plot(
+    n_molecules: int,
+    fingerprints: list,
+    times: list,
+    save: bool = True,
+    type: str = "time",
+) -> None:
+    fig = plt.figure(figsize=(15, 10))
+    ax1 = fig.add_subplot()
+    ax1.set_ylabel("Fingerprints")
+    fp_names = [fp.__name__ for fp in fingerprints]
+
+    if type == "time":
+        file_name = "times_of_sequential_computation"
+        ax1.set_xlabel("Time of computation")
+        ax1.set_title("Times of sequential computation for all fingerprints")
```

Comment:
`Sequential computation time`. The `for all fingerprints` part is not necessary for any plot